core/src/main/java/nu/yona/server/subscriptions/service/PinResetRequestService.java

```diff
@@ -0,0 +1,81 @@
+package nu.yona.server.subscriptions.service;
+
+import java.time.ZonedDateTime;
+import java.util.UUID;
+
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import nu.yona.server.exceptions.PinResetRequestConfirmationException;
+import nu.yona.server.properties.YonaProperties;
+import nu.yona.server.subscriptions.entities.ConfirmationCode;
+import nu.yona.server.subscriptions.entities.User;
+
+@Service
+public class PinResetRequestService
+{
+	@Autowired
+	private UserService userService;
+
+	@Autowired
+	private YonaProperties yonaProperties;
+
+	@Transactional
+	public void requestPinReset(UUID userID)
+	{
+		ConfirmationCode confirmationCode = createConfirmationCode();
```

Comment:
This should have a check that there does not exist a pin reset request yet (see remark about test). Otherwise it is stil possible to bruteforce the code the other way around.